dashboard/modules/reporter/reporter_agent.py

```diff
@@ -465,10 +465,13 @@ def _get_workers(self):
             return []
         else:
             workers = set(raylet_proc.children())
+            # Must keep the original self._workers in case of psutil.cpu_percent is 0
+            # See more: https://github.com/ray-project/ray/issues/29848
+            self._workers.intersection_update(workers)
+            self._workers.update(workers)
             # Remove the current process (reporter agent), which is also a child of
             # the Raylet.
-            workers.discard(psutil.Process())
-            self._workers = workers
+            self._workers.discard(psutil.Process())
```

Comment:
Just to be clear when as_dict is called, it refreshes the values right? Can you double check?

Reply:
1. I believe the `as_dict` itself will not **refreshes** or get the correct CPU percent the first time `_get_worker()` is called. It just reads the property of `Process()`. This should work because `self._workers` is consistent and when it should show the correct metric in the following calls of `_get_worker()` (rather than the `as_dict`).
2. I have tested the script in the original issue (#29848) and it looks fine. 
![image](https://user-images.githubusercontent.com/15307797/199013161-4169f479-9147-42ff-8605-de722c5e62fe.png)

Reply:
Makes sense! 

For 2, can you add a test (under modules/reporter/test/) to avoid having the same issue in the future?

Reply:
> For 2, can you add a test (under modules/reporter/test/) to avoid having the same issue in the future?

OK, I will try and will add comments here if any problems are encountered.